pallets/liquidity-mining/src/lib.rs

```diff
@@ -0,0 +1,754 @@
+	GlobalPool<Period, Balance, AssetId, AccountId, BlockNumber>
+{
+	fn new(
+		id: PoolId,
+		updated_at: Period,
+		reward_currency: AssetId,
+		yield_per_period: Permill,
+		planned_yielding_periods: Period,
+		blocks_per_period: BlockNumber,
+		owner: AccountId,
+		incentivized_token: AssetId,
+		max_reward_per_period: Balance,
+	) -> Self {
+		Self {
+			accumulated_rewards: Default::default(),
+			accumulated_rps: Default::default(),
+			accumulated_rps_start: Default::default(),
+			paid_accumulated_rewards: Default::default(),
+			total_shares: Default::default(),
+			id,
+			updated_at,
+			reward_currency,
+			yield_per_period,
+			planned_yielding_periods,
+			blocks_per_period,
+			owner,
+			incentivized_token,
+			max_reward_per_period,
+		}
+	}
+}
+
+#[derive(Clone, Encode, Decode, PartialEq, Eq, RuntimeDebug, TypeInfo)]
+pub struct LiquidityPool<Period, Balance> {
+	id: PoolId,
+	updated_at: Period,
+	total_shares: Balance,
+	accumulated_rps: Balance,
+	loyalty_curve: Option<LoyaltyCurve>,
+	stake_in_global_pool: Balance,
+	multiplier: u32, //this is multiplier/weight in glboal pool
+}
+
+impl<Period, Balance: std::default::Default> LiquidityPool<Period, Balance> {
+	fn new(id: PoolId, updated_at: Period, loyalty_curve: Option<LoyaltyCurve>, multiplier: u32) -> Self {
+		Self {
+            accumulated_rps: Default::default(),
+            stake_in_global_pool: Default::default(),
+            total_shares: Default::default(),
+			id,
+			updated_at,
+			loyalty_curve,
+			multiplier,
+		}
+	}
+}
+
+#[derive(Clone, Encode, Decode, PartialEq, Eq, RuntimeDebug, TypeInfo)]
+pub struct LoyaltyCurve {
+	b: FixedU128,
+	scale_coef: u32,
+}
+
+impl Default for LoyaltyCurve {
+	fn default() -> Self {
+		Self {
+			b: FixedU128::from_inner(500_000_000_000_000_000), // 0.5
+			scale_coef: 100,
+		}
+	}
+} use frame_support::pallet_prelude::*;
+use frame_support::sp_runtime::{traits::AtLeast32BitUnsigned, FixedPointNumber, FixedPointOperand};
+use sp_std::convert::{From, Into, TryInto};
+
+use primitives::asset::AssetPair;
+
+#[frame_support::pallet]
+pub mod pallet {
+	use super::*;
+	use crate::weights::WeightInfo;
+	use frame_system::pallet_prelude::{BlockNumberFor, OriginFor};
+
+	#[pallet::pallet]
+	pub struct Pallet<T>(_);
+
+	#[pallet::hooks]
+	impl<T: Config> Hooks<BlockNumberFor<T>> for Pallet<T> {}
+
+	#[pallet::config]
+	pub trait Config: frame_system::Config {
+		type Event: From<Event<Self>> + IsType<<Self as frame_system::Config>::Event>;
+
+		/// Balance type
+		type Balance: Parameter
+			+ From<u128>
+			+ Into<u128>
+			+ From<Self::BlockNumber>
+			//+ From<FixedU128>
+			+ Member
+			+ AtLeast32BitUnsigned
+			+ FixedPointOperand
+			+ Default
+			+ Copy
+			+ MaybeSerializeDeserialize
+			+ MaxEncodedLen
+			+ CheckedAdd;
+
+		/// Asset type
+		type CurrencyId: Parameter + Member + Copy + MaybeSerializeDeserialize + Ord + From<u32>;
+
+		/// Currency for transfers
+		type MultiCurrency: MultiCurrencyExtended<
```

Comment:
MultiCurrency should be usually sufficient .

Extended only provides `update_balance` which i believe it is not needed